src/cli/iohandlers/zonefile_handler.go

```diff
@@ -0,0 +1,155 @@
+/*
+ * ZDNS Copyright 2024 Regents of the University of Michigan
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package iohandlers
+
+import (
+	"bufio"
+	"context"
+	"os"
+	"strings"
+	"sync"
+
+	log "github.com/sirupsen/logrus"
+
+	"github.com/zmap/zdns/v2/src/internal/util"
+)
+
+type ZoneFileInputHandler struct {
+	filepath       string
+	includeTargets bool
+}
+
+func NewZoneFileInputHandler(filepath string, includeTargets bool) *ZoneFileInputHandler {
+	return &ZoneFileInputHandler{
+		filepath:       filepath,
+		includeTargets: includeTargets,
+	}
+}
+
+// parseZoneLine parses a zone file record and returns domains to query.
+func (h *ZoneFileInputHandler) parseZoneLine(line string) []string {
+	line = strings.TrimSpace(line)
+
+	if line == "" || strings.HasPrefix(line, ";") {
+		return nil
+	}
+
+	fields := strings.Fields(line)
+	if len(fields) < 3 {
+		return nil
+	}
+
+	domains := make([]string, 0, 2)
+
+	name := strings.ToLower(fields[0])
+	name = strings.TrimSuffix(name, ".")
+	if name != "" {
+		domains = append(domains, name)
+	}
+
```

Comment:
```suggestion
if !h.includeTargets {
    return domains
}
```
Believe we can return early here and save the other field parsing if we don't need to include targets as well.